### DISCUSSION TITLE
SSR - don't include SSR rendered component JS

### DISCUSSION THREAD

**Discussion by adamlewkowicz** [vercel/next.js#17851] [Help] [answered]
I would like to render the HTML of the component on the server-side, that will be loaded on the client only when it comes to the viewport.
The idea is generally to render the HTML on SSR and import the component's code only when visible using Intersection Observer.

```tsx
if (process.browser) {
  // Render if visible on the client
  if (isVisible) {
    return <ButtonLazy />
  }
}

// Render as default on the server - but don't include JS in a bundle
return <ButtonLazy />
```
Is it somehow possible to configure `dynamic()` method options not to include JS of SSR's loaded component?

**Comment by jamesmosier** [comment, marked as answer]
That is not currently possible with `next/dynamic` and I don't believe it is possible in general. You'd then have a mismatch between the client and server and React would not know what to hydrate the client with.
